lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -607,3 +662,18 @@ void llbuild::buildsystem::setDefaultQualityOfService(QualityOfService level) {
   defaultQualityOfService = level;
 }
 
+SchedulerAlgorithm llbuild::buildsystem::getSchedulerAlgorithm() {
+  return defaultAlgorithm;
+}
+
+void llbuild::buildsystem::setSchedulerAlgorithm(SchedulerAlgorithm algorithm) {
+  defaultAlgorithm = algorithm;
+}
+
+uint32_t llbuild::buildsystem::getSchedulerLaneWidth() {
+  return defaultLaneWidth;
+}
+
+void llbuild::buildsystem::setSchedulerLaneWidth(uint32_t width) {
+  defaultLaneWidth = width;
```

Comment:
I wonder if we should impose a limit here? Unlike systems like say Ninja which uses a select-based mechanism, our lane based approach isn't very compatible with setting a super high lane count... OTOH, its hard to pick a sane limit, so maybe we should just say caveat emptor.

Reply:
Caveat emptor it is